src/ucs/datastruct/pgtable.c

```diff
@@ -614,6 +614,11 @@ void ucs_pgtable_purge(ucs_pgtable_t *pgtable, ucs_pgt_search_callback_t cb,
     ucs_status_t status;
     unsigned i;
 
+    if (num_regions == 0) {
```

Comment:
why is this considered a problem? because of zero-size allocation?

Reply:
Yes. 
Memory debugger (Electric Fence) reports issue on calling `calloc` with number of elements equal to zero.

Reply:
btw, it also could print a `ucs_warn` in case of `num_regions == 0` which is not expected

Reply:
why not? we can have empty rcache
btw, would the memory debugger report an issue if we used malloc instead of calloc?

Reply:
I think that warning level is too high in this case.
Purging a page table with zero number of regions may occur in case of an issue at the external level. E.g. an object creates table, and immediately purges the table before adding a region, because smth went wrong. Thus, the warning/error should be reported at the external level. And here debug is enough, IMO.

Reply:
>> btw, would the memory debugger report an issue if we used malloc instead of calloc?
Yes, it would.